### PR TITLE
Fix bug:missing submap key, add dart tests

### PR DESCRIPTION
**Pull request by johncrim**
Issue described here:
https://github.com/oddbird/accoutrement/issues/49

This is still an issue with dart-sass when `$handle-missing-keys` is not set. PR includes a test case for this, which failed with previous code on dart-sass only.

PR also adds most of the existing tests to run in dart-sass. There is more work to do to make all tests run in dart-sass, but that is outside the scope of this issue.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "test": "mocha",
     "lint": "gulp sasslint",
     "docs": "gulp sassdoc",
-    "commit": "yarn test; yarn lint; yarn docs"
+    "commit": "yarn test && yarn lint && yarn docs"
```
Comment (johncrim) [inline package.json:30]: This is needed to run `yarn commit` on Windows. I believe it should still work on Mac + Linux.

File: package.json [hunk 2]
```diff
@@ -39,7 +39,8 @@
     "gulp": "^4.0.2",
     "gulp-stylelint": "^13.0.0",
     "mocha": "^8.0.1",
-    "node-sass": "^4.14.1",
+    "node-sass": "^5.0.0",
```
Comment (johncrim) [inline package.json:42]: The previous version of node-sass had some outdated dependencies (node-gyp), and I thought it best to build using current node-sass. I can revert this if necessary.

File: test/core/_api.scss [hunk 3]
```diff
@@ -94,7 +94,22 @@
   @include it('Fails elegantly when keys are not found') {
     @include assert-equal(
       (get-token($map, 'dop'), get-token($map, 'dip')),
-      ('("zab": 4)->nope', '#that->baz')
+      ('#ofo->nope', '#that->baz')
+    );
+  }
+
+  @include it('Can return submaps') {
```
Comment (johncrim) [inline test/core/_api.scss:101]: This failed before the fix, on dart-sass only.

File: sass/core/_strings.scss [hunk 4]
```diff
@@ -71,8 +71,16 @@
...
   } @else if $i {
-    // Force $new & $string to string
-    $new: if((type-of($new) == 'string') or (not $new), $new, '#{$new}');
+    // Convert $new to a string
+    @if ($new and (type-of($new) != 'string')) {
+      @if (type-of($new) == 'map') {
+        @warn '_a_str-replace $new parameter cannot be a map';
+        @return $string;
+      }
+      @else {
```
Comment (mirisuzanne) [inline sass/core/_strings.scss:80]:
```suggestion
      } @else {
```

just for formatting consistency…

**Review by mirisuzanne — COMMENTED** [review]
looks good! Thanks!

Can you update the changelog as well? I think it probably makes sense to just do a major release, and get this out.

**Comment by johncrim** [comment]
Sure, I'll update the changelog (@mirisuzanne - assuming you're asking me to do it). I don't know about the release process, let me know if that's a request for me (and I'll need docs on the process) or someone else.

**Comment by johncrim** [comment]
I'm going to hold off on updating the changelog until I hear whether you want me to do it, or a maintainer (@jgerigmeyer ?). I'm happy to help out, but don't want to step on anyone's toes.

**Comment by mirisuzanne** [comment]
@johncrim Go for it. You can add a new block at the top of the markdown file, with `UNRELEASED` instead of a version number and date. Then follow the pattern established there – a list-item describing the change, prefixed with e.g. `BREAKING`, the relevant `(#issue)`. Does that make sense?

**Comment by johncrim** [comment]
Changelog has been updated. My other PR (#62) builds on top of this one, b/c it needs the dart tests. So please merge this one first.

**Review by mirisuzanne — APPROVED** [review]

**Comment by mirisuzanne** [comment]
Thanks @johncrim !

**Pull request merged**